Marketplace/MarketplaceToolkit.ps1

```diff
@@ -0,0 +1,2109 @@
+                                            <Label Content="API endpoint" />
+                                        </StackPanel>
+                                        <StackPanel Orientation="Horizontal" Width="258" Height="23" Margin="0,0,0,8" ToolTipService.InitialShowDelay="0" ToolTipService.ShowDuration="2147483647" >
+                                            <TextBox x:Name="Publish_Tbx_Endpoint" Width="258" />
+                                            <Button Width="15" Content="!" Foreground="White" Background="Red" BorderThickness="1" BorderBrush="Red" Visibility="Collapsed" Cursor="Hand" />
+                                        </StackPanel>
+
+                                        <Label Content="Publishing log" Margin="0,10,0,10" />
+                                        <ListView x:Name="Publish_Lsv_Log" MinHeight="23" Background="#F9F9F9F9"/>
+                                        <Border BorderBrush="#CCCCCC" BorderThickness="0,1,0,0" Margin="0,15,0,0">
+                                            <Button x:Name="Publish_Btn_Publish" Width="60" Height="23" Content="Publish" HorizontalAlignment="Right" Margin="0,15,0,0"></Button>
+                                        </Border>
+                                    </StackPanel>
+                                </ScrollViewer>
+                            </Grid>
+                        </Border>
+                        <!--#endregion Blade_Upload -->
+                    </DockPanel>
+                </ScrollViewer>
+            </Grid>
+        </DockPanel>
+    </Grid>
+</Window>
+'@
+#endregion
+
+#region Get XAML and create variables
+Add-Type -AssemblyName PresentationFramework
+
+[XML]$XML = $XAML -replace "x:N",'N' 
+
+$Reader = (New-Object System.XML.XMLNodeReader $XML)
+$Form = [Windows.Markup.XAMLReader]::Load($Reader)
+
+# Create variables > Load XAML Objects In PowerShell
+$XML.SelectNodes("//*[@Name]") | % {Set-Variable -Name "X_$($_.Name)" -Value $Form.FindName($_.Name)}
+
+$timer = new-object 'System.Windows.Threading.DispatcherTimer'
+$timer.Interval = [TimeSpan]"0:0:1.00"
+
+#endregion
+
+#region Functions
+function F_Regex {
+$X_Input_Tbx_Name.Tag = @{'regex'='[ a-zA-Z0-9]{1,80}';'errormessage'='Maximum 80 characters. Can only contain A-Z, a-z, 0-9 and spaces.'}
+$X_Input_Tbx_Publisher.Tag = @{'regex'='[ a-zA-Z0-9]{1,20}';'errormessage'='Maximum 30 characters. Can only contain A-Z, a-z, 0-9 and spaces.'}
+$X_Input_Tbx_Summary.Tag = @{'regex'='[ a-zA-Z0-9]{1,100}';'errormessage'='Maximum 100 characters. Can only contain A-Z, a-z, 0-9 and spaces.'}
+$X_Input_Tbx_Description.Tag = @{'regex'='.{1,5000}';'errormessage'='Maximum 5000 characters.'}
+$X_Input_Tbx_Category.Tag = @{'regex'='[- a-zA-Z0-9]{1,64}';'errormessage'='Maximum 64 characters. Can only contain A-Z, a-z, 0-9, - and spaces.'}
+$X_Input_Tbx_Icon40.Tag = @{'errormessage'='The image size must be 40x40.'}
```

Comment:
Usually use hard code strings for error messages is not perfect. 
1. we may need to have review by tech writer and it can be convenient to have them in a central place
2. possibly some day in the future, the UI needs to have localization